Release 0.6.0

mhei opened the pull request:
This PR includes:
- improve support for Charge Control Y
- add possibility to read out flash and to verify during flashing
- add migration for changed parameter blocks
- YAML tools for parameter block files

mhei committed:
Add first safety firmware for parsley platform

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Install Parsley param block/fw and streamline filename for Charge SOM

This already includes the changed special value for ununsed PT1000 channels.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Adapt magic value for unused PT1000 channel

Upcoming firmware versions expect the new magic value as indication for
unused channels.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Add support for new fields in CS2

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-raw: decide mcs mode also on receive Charge State frame

It can happen, that we receive a Charge State frame before our
inquiry for firmware version responded, so handle this too.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: add dump command to read flash content

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: drop superfluous RWE command

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: add verify to flash procedure

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Update to latest safety firmwares

For Charge SOM:
- fix temperature reading issues

For Parsley:
- fix for instable CE state
- report disable PT1000 channels correctly

For both:
- changed magic value for disabled PT1000 channel in parameter block
  -> needs migration from 0x8000 to 0x1fff

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: use dash in fw-info instead of underscore to align commands

Implement a migration helper which accepts the underscore variant as well.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: fix fw-info command

We need to map the file also when fw-info is used with filename,
so let's invert the check and prevent the file mapping when dumping.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Factor out some parameter related stuff

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-gen-param-block: fix newlines on error messages

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Introduce ra-pb-dump helper tool

This dumps a parameter block as YAML structure, either from a file or stdin.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
param_block: handle also older value for 'disabled'

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Print optional timestamp when dumping frames

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-dump: add example usage and finetune messages

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Allow more string to express disabled parameters

The idea is to relax the stuff we expect and thus making it more
human-intuitive, but during output of YAML, we are strict and
print it with canonical strings.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Introduce ra-pb-create - cmdline tool to convert YAML to parameter block

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-raw: add reset pin handling

The reset pin is pulled by default on tool startup, but it can
be prevented by giving a command line parameter.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
cb_uart: add missing frame types

Add at least all other frame types so that debug messages are
more speaking instead of jump dumping UNKNOWN.
Most frames are not required for daily usage, so we wont implement
them, but at least traces should give a hint when receiving.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
README.md: mention new tools

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-create: small usability/maintainance fixes

- align variable name with others
  Looks that editor did some internal magic and I did not notice.
- fix define name
- add checks and warnings in case YAML does not specify all possible
  parameters

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
param_block: use dump string comparison

We expect UTF-8 YAML file and also code is and case-senstive is
not required here since uncommon, so let's use dump strcmp().

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Factor out parameter block file initialization and crc update

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
param_block: fix disable magic endianess

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-update: add migration for changed magic value

The magic value in the parameter binary changed. So when updating
from version 0.2.2 to a newer one, let's dump and re-install the
parameter block which causes the magic value to be adapted for
the new firmware version.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-gen-param-block: fix backwards compatibility

Relax the checks and allow more values for disabled channels
to be backwards compatible again.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-create: add explicite error for wrong YAML file encoding

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Add safe state reason stuff for CS1 frame

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Align spelling of safe state -> safestate

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Update parsley firmware to latest development release

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Bump version to 0.6.0

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
cb_proto_cs2_get_safe_state_active: fix typo in offset

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

lategoodbye commented:
In general i'm fine with the pull request.

Regarding the new ra-pb-create tool, i like to have some example YAML files. How about storing the "original" YAML files of chargeSOM_parameter-block_only-contactor.bin and parsley_parameter-block_all-pt1000-disabled.bin in this repo and generate the bin files during build?

mhei commented:
I like the idea of getting rid of the binaries here in the repo. I think, it might even be better do adapt the update script to use the new YAML file as base and generating the required bin file on the fly.

mhei committed:
Use YAML parameter block files in favor of binary ones

It is easier for diffing, and Human-readable too - so let's switch
the binary files to YAML files.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>
